File: nanopdf-js/examples/01-basic-reading.ts
/**
 * Example 1: Basic PDF Reading
 *
 * This example demonstrates how to:
 * - Open a PDF document
 * - Read basic properties
 * - Access metadata
 * - Load and inspect pages
 */

import { Document } from 'nanopdf';
import { resolve } from 'path';

function basicReading() {
  // Open a PDF document
  const pdfPath = resolve(__dirname, '../../test-pdfs/simple/hello-world.pdf');
  const doc = Document.open(pdfPath);

  console.log('=== Basic PDF Information ===');
  console.log(`File: ${pdfPath}`);
  console.log(`Format: ${doc.format}`);
  console.log(`Pages: ${doc.pageCount}`);
  console.log(`Needs Password: ${doc.needsPassword()}`);
  console.log(`Is Authenticated: ${doc.isAuthenticated}`);

  // Read metadata
  console.log('\n=== Metadata ===');
  const metadataKeys = ['Title', 'Author', 'Subject', 'Keywords', 'Creator', 'Producer'];
  for (const key of metadataKeys) {
    const value = doc.getMetadata(key);
    if (value) {
      console.log(`${key}: ${value}`);
    }
  }

  // Load and inspect the first page
  console.log('\n=== First Page ===');
  const page = doc.loadPage(0);
  console.log(`Page Number: ${page.pageNumber}`);
  console.log(`Bounds: [${page.bounds.x0}, ${page.bounds.y0}, ${page.bounds.x1}, ${page.bounds.y1}]`);
  console.log(`Width: ${page.bounds.width} points`);
  console.log(`Height: ${page.bounds.height} points`);
  console.log(`Rotation: ${page.rotation} degrees`);

  // Convert points to inches (72 points = 1 inch)
  const widthInches = page.bounds.width / 72;
  const heightInches = page.bounds.height / 72;
  console.log(`Size: ${widthInches.toFixed(2)}" × ${heightInches.toFixed(2)}"`);

  // Clean up
  page.drop();
  doc.close();

  console.log('\n✅ Done!');
}

// Run the example
if (require.main === module) {
  try {
    basicReading();
  } catch (error) {
    console.error('Error:', error);
    process.exit(1);
  }
}

export { basicReading };

File: go-micropdf/examples/01_basic_reading.go
// Example 1: Basic PDF Reading
//
// This example demonstrates how to:
//   - Open a PDF document
//   - Read basic properties
//   - Access metadata
//   - Load and inspect pages

//go:build example

package main

import (
	"fmt"
	"log"
	"path/filepath"

	micropdf "github.com/lexmata/micropdf/go-micropdf"
)

func main() {
	// Create context (required for all PDF operations)
	ctx := micropdf.NewContext()
	if ctx == nil {
		log.Fatal("Failed to create context")
	}
	defer ctx.Drop()

	// Open a PDF document
	pdfPath := filepath.Join("..", "..", "test-pdfs", "simple", "hello-world.pdf")
	doc, err := micropdf.OpenDocument(ctx, pdfPath)
	if err != nil {
		log.Fatalf("Failed to open PDF: %v", err)
	}
	defer doc.Drop()

	fmt.Println("=== Basic PDF Information ===")
	fmt.Printf("File: %s\n", pdfPath)

	// Get page count
	pageCount, err := doc.PageCount()
	if err != nil {
		log.Fatalf("Failed to get page count: %v", err)
	}
	fmt.Printf("Pages: %d\n", pageCount)

	// Check if password is needed
	needsPassword, err := doc.NeedsPassword()
	if err != nil {
		log.Printf("Warning: Could not check password status: %v", err)
	} else {
		fmt.Printf("Needs Password: %v\n", needsPassword)
	}

	// Read metadata
	fmt.Println("\n=== Metadata ===")
	metadataKeys := []string{"Title", "Author", "Subject", "Keywords", "Creator", "Producer"}
	for _, key := range metadataKeys {
		value, err := doc.GetMetadata(key)
		if err == nil && value != "" {
			fmt.Printf("%s: %s\n", key, value)
		}
	}

	// Load and inspect the first page
	fmt.Println("\n=== First Page ===")
	page, err := doc.LoadPage(0)
	if err != nil {
		log.Fatalf("Failed to load page: %v", err)
	}
	defer page.Drop()

	pageNum := page.PageNumber()
	fmt.Printf("Page Number: %d\n", pageNum)

	bounds := page.Bounds()
	fmt.Printf("Bounds: [%.2f, %.2f, %.2f, %.2f]\n",
		bounds.X0, bounds.Y0, bounds.X1, bounds.Y1)
	fmt.Printf("Width: %.2f points\n", bounds.Width())
	fmt.Printf("Height: %.2f points\n", bounds.Height())

	// Convert points to inches (72 points = 1 inch)
	widthInches := bounds.Width() / 72
	heightInches := bounds.Height() / 72
	fmt.Printf("Size: %.2f\" × %.2f\"\n", widthInches, heightInches)

	fmt.Println("\n✅ Done!")
}
